OpenScienceJournal/whistlepunk_library/src/main/java/com/google/android/apps/forscience/ble/MkrSciBleManager.java

```diff
@@ -0,0 +1,372 @@
+                mGatt = gatt;
+                mCharacteristics.clear();
+                mGatt.discoverServices();
+            } else if (newState == BluetoothProfile.STATE_DISCONNECTED) {
+                mReadyForAction = false;
+                gatt.disconnect();
+            }
+        }
+
+        @Override
+        public void onServicesDiscovered(BluetoothGatt gatt, int status) {
+            BluetoothGattService service = gatt.getService(UUID.fromString(SERVICE_UUID));
+            if (service != null) {
+                mCharacteristics.addAll(service.getCharacteristics());
+            }
+            mReadyForAction = true;
+            onGattActionCompleted();
+        }
+
+        @Override
+        public void onDescriptorRead(BluetoothGatt gatt, BluetoothGattDescriptor descriptor, int status) {
+            onGattActionCompleted();
+        }
+
+        @Override
+        public void onDescriptorWrite(
+                BluetoothGatt gatt, BluetoothGattDescriptor descriptor, int status) {
+            onGattActionCompleted();
+        }
+
+        @Override
+        public void onCharacteristicRead(
+                BluetoothGatt gatt, BluetoothGattCharacteristic characteristic, int status) {
+            onGattActionCompleted();
+        }
+
+        @Override
+        public void onCharacteristicWrite(BluetoothGatt gatt, BluetoothGattCharacteristic characteristic, int status) {
+            onGattActionCompleted();
+        }
+
+        @Override
+        public void onCharacteristicChanged(
+                BluetoothGatt gatt, BluetoothGattCharacteristic characteristic) {
+            final String uuid = characteristic.getUuid().toString();
+            final ValueType type;
+            switch (uuid) {
+                case INPUT_1_UUID:
+                case INPUT_2_UUID:
+                case INPUT_3_UUID:
+                    type = ValueType.UINT16;
+                    break;
+                case VOLTAGE_UUID:
+                case CURRENT_UUID:
+                case RESISTANCE_UUID:
+                    type = ValueType.SFLOAT;
+                    break;
+                case ACCELEROMETER_UUID:
+                case GYROSCOPE_UUID:
+                case MAGNETOMETER_UUID:
+                    type = ValueType.SFLOAT_ARR;
+                    break;
+                default:
+                    type = null;
+            }
+            if (type != null) {
+                final double[] values = parse(type, characteristic.getValue());
+                if (values != null) {
+                    // filter to avoid too large values blocking the UI
+                    for (int i = 0; i < values.length; i++) {
+                        if (values[i] > MAX_VALUE) {
+                            values[i] = MAX_VALUE;
+                        } else if (values[i] < MIN_VALUE) {
+                            values[i] = MIN_VALUE;
+                        }
+                        /*
```

Comment:
What's this commented out code for? Can you either remove it or add a comment that explains why it is here?

Reply:
Removed, it was just debug code.